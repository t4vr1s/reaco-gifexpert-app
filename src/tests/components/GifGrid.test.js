import '@testing-library/jest-dom'
import React from 'react'
import { shallow } from 'enzyme'
import GifGrid from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'

/* eslint-disable */
jest.mock('../../hooks/useFetchGifs')

describe('Pruegas al componente < GifGrid />', () => {
  const category = 'One Punch Man'

  test('Snapshot inicial del componente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })

    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should mostrar los items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [
      {
        id: 'abc',
        url: 'https://localhost/ruta/imagen.png',
        title: 'titulo de test'
      },
      {
        id: 'def',
        url: 'https://localhost/ruta/imagen-nueva.png',
        title: 'Imagen nueva'
      }
    ]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  })
})
