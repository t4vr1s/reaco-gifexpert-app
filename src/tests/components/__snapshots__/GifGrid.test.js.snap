// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruegas al componente < GifGrid /> Snapshot inicial del componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch Man
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
